core/src/test/java/org/elasticsearch/index/query/TemplateQueryBuilderTest.java

```diff
@@ -18,21 +18,41 @@
  */
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.search.Query;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentFactory;
 import org.elasticsearch.script.ScriptService.ScriptType;
 import org.elasticsearch.script.Template;
-import org.elasticsearch.test.ESTestCase;
 import org.junit.Test;
 
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
 
-/**
- * Test building and serialising a template search request.
- * */
-public class TemplateQueryBuilderTest extends ESTestCase {
+import static org.hamcrest.Matchers.is;
+
+public class TemplateQueryBuilderTest extends BaseQueryTestCase<TemplateQueryBuilder> {
+
+    @Override
+    protected boolean supportsBoostAndQueryName() {
+        return false;
+    }
+
+    @Override
+    protected TemplateQueryBuilder doCreateTestQueryBuilder() {
+        return new TemplateQueryBuilder(new Template(RandomQueryBuilder.createQuery(random()).toString()));
+    }
+
+    @Override
+    protected void doAssertLuceneQuery(TemplateQueryBuilder queryBuilder, Query query, QueryShardContext context) throws IOException {
+        assertEquals(query, queryBuilder.toQuery(context));
```

Comment:
this assertion makes little sense. we call queryBuilder.toQuery, then verify that the result is the same that we obtain calling queryBuilder.toQuery again. This will always be true but it's not a meaningful test. I'd rather leave the method empty to be honest if we don't know what to test and how. Maybe we should instead change the test query and pass in an actual template with params, rather than a random query?